Automattic/vip-go-mu-plugins
# Adds corrupted version detection

## Description

Self-healing versioning options

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

Outline the steps to test and verify the PR here.

1. Break your versioning options by running - `lando wp eval "update_option( 'vip_search_index_versions', 'malformed' );"`
1. Run `wp vip-search index-versions list post` shows correct versions (with lowest version active)
1. Add version 
1. Remove option  `lando wp eval "delete_option('vip_search_index_versions');"`
1. Run `wp vip-search index-versions list post` shows correct versions (with lowest version active)


Note:
To clean cache lock in order to test:
`lando wp eval "wp_cache_set('index_versions_self_heal_lock', 0, 'vip_search');"`


## Changelog 

###Search Corrupted Index Version Detection

Adding a feature that detects if index versions on given site are in incorrect state or missing.

Next step is to automatically recreate the versioning information by scanning existing ES indexes. 

At this point the change only detects an issue and logs the state it would recreate (without actually doing it). We will review those changes first and then enable self-healing.

https://github.com/Automattic/vip-go-mu-plugins/pull/1881
